chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java

```diff
@@ -466,6 +468,72 @@ public QuestionableActivityListingDTO checkRealWorldTestingResultsRemoved(
         return activity;
     }
 
+    public List<QuestionableActivityListingDTO> checkMeasuresAdded(
+            CertifiedProductSearchDetails origListing, CertifiedProductSearchDetails newListing) {
+
+        List<QuestionableActivityListingDTO> measuresAddedActivities = new ArrayList<QuestionableActivityListingDTO>();
+        if (origListing.getMeasures() != null && origListing.getMeasures().size() > 0
+                && newListing.getMeasures() != null && newListing.getMeasures().size() > 0) {
+            for (ListingMeasure newMeasure : newListing.getMeasures()) {
+                Optional<ListingMeasure> matchingOrigMeasure = origListing.getMeasures().stream()
+                    .filter(origMeasure -> origMeasure.getId().equals(newMeasure.getId()))
+                    .findFirst();
+                if (!matchingOrigMeasure.isPresent()) {
+                    QuestionableActivityListingDTO activity = new QuestionableActivityListingDTO();
+                    activity.setBefore(null);
+                    activity.setAfter(newMeasure.getMeasurementType().getName()
+                            + " measure " + newMeasure.getMeasure().getName()
+                            + " for " + newMeasure.getMeasure().getAbbreviation());
+                    measuresAddedActivities.add(activity);
+                }
+            }
+        } else if (newListing.getMeasures() != null
+                && (origListing.getMeasures() == null || origListing.getMeasures().size() == 0)) {
+            for (ListingMeasure newMeasure : newListing.getMeasures()) {
+                    QuestionableActivityListingDTO activity = new QuestionableActivityListingDTO();
+                    activity.setBefore(null);
+                    activity.setAfter(newMeasure.getMeasurementType().getName()
+                            + " measure " + newMeasure.getMeasure().getName()
+                            + " for " + newMeasure.getMeasure().getAbbreviation());
+                    measuresAddedActivities.add(activity);
+            }
+        }
+        return measuresAddedActivities;
+    }
+
+    public List<QuestionableActivityListingDTO> checkMeasuresRemoved(
+            CertifiedProductSearchDetails origListing, CertifiedProductSearchDetails newListing) {
+
+        List<QuestionableActivityListingDTO> measuresRemovedActivities = new ArrayList<QuestionableActivityListingDTO>();
+        if (origListing.getMeasures() != null && origListing.getMeasures().size() > 0
+                && newListing.getMeasures() != null && newListing.getMeasures().size() > 0) {
+            for (ListingMeasure origMeasure : origListing.getMeasures()) {
+                Optional<ListingMeasure> matchingNewMeasure = newListing.getMeasures().stream()
+                    .filter(newMeasure -> origMeasure.getId().equals(newMeasure.getId()))
+                    .findFirst();
```

Comment:
Just an FYI...  `findAny` is more efficient than `findFirst` since streams are not necessarily processing in any order.
https://www.baeldung.com/java-stream-findfirst-vs-findany

Reply:
Addressed